cmd/git_utils.go

```diff
@@ -129,20 +139,59 @@ func GetGitInfo(config *RootCommandConfig) (GitInfo, error) {
 		}
 
 	}
+
 	if strings.Contains(value, noCommits) {
 		gitInfo.Branch = StringAfter(value, noCommits)
 	}
+
 	changesMade := false
-	outputLength := len(outputLines)
+	outputLength := len(statusOutputLines)
 
 	if outputLength > 1 {
 		changesMade = true
-
 	}
 	gitInfo.ChangesMade = changesMade
 
-	errMsg := ""
+	if gitInfo.Upstream == "" {
+		config.Debug.log("Failed to retrieve upstream from git status -sb. Looking via latest commit")
+		outputLines, err := RunGitBranchContains(config.LoggingConfig, gitInfo.Commit.SHA, config.ProjectDir, lineSeparator, config.Dryrun)
```

Comment:
After running `git status -sb`, we only know that this branch is tracking an remote upstream branch. We do not know that the commit at the head of the local branch is in fact in the remote branch. I suggest that we always run `git branch contains`, and in the case where the upstream has already been set we verify that it is outputted by the command. If the upstream hasn't been set by `git status -sb`, we continue as per https://github.com/appsody/appsody/pull/959/files#diff-5675d25a846ed226e9ba4514bd7eef60R161-R168